### ISSUE TITLE
Switch to gRPC API

### ISSUE THREAD

**Issue by peburrows** [peburrows/kane#5]
Once we have a decent http/2 client library & protocol buffer support added, we should switch to gRPC.

**Comment by peburrows** [comment]
Once [River](https://github.com/peburrows/river) is ready, we'll use that as the basis for our gRPC support.

**Comment by barthez** [comment]
Hey @peburrows, have you thought about using https://github.com/ericmj/mint as HTTP/2 client?

**Comment by peburrows** [comment]
Mint was only recently release, but it's definitely an option. I haven't yet had the chance to look into it too deeply to know whether or not it would be a good fit, though.

**Comment by xingxing** [comment]
@peburrows  our team have worked with https://github.com/elixir-grpc/grpc for both grpc server and client side, it seems worked well for us.

**Comment by brianbowden** [comment]
It's a little disappointing that gRPC support is not yet present in Kane. I've been using [Weddell](https://github.com/cjab/weddell) for a while now with good results.

**Comment by peburrows** [comment]
@brianbowden yeah, I don't disagree; unfortunately, my priorities have just been elsewhere recently. This is still an issue I absolutely want to address, I just haven't really had the time.

That said, I'll look to dig into this in the near future. As always, contributions are also welcome.

**Comment by brianbowden** [comment]
@peburrows That said, I appreciate all the work you all have done! I use goth all the time.

**Comment by kentbull** [comment]
@brianbowden have you been able to get high transaction volumes through rapidly with Weddell? I'm seeing 1-3 second publishing delays and about 1.5 second pull delays. It's kind of crazy. I've got to be using it wrong 🤦🏻
